# Cleansing Data Youtube Rewind 2016

#### Import Library

In [1]:
from dotenv import load_dotenv
from pymongo import MongoClient
import os
import pandas as pd

#### Import url dari .env

In [2]:
# Memuat value dari file .env
load_dotenv()

mongodb_url = os.getenv('URL_SANDY')
local_url = os.getenv('URL_LOCAL')

In [3]:
database_name = "youtube_rewind_indonesia"
collection_from = "collect_2016"
collection_name = "cleansing_2016"

#### Import dataset dari MongoDB

In [4]:
# Koneksi ke MongoDB
client = MongoClient(local_url)

# Pilih database dan koleksi
db = client[database_name]
collection = db[collection_from]

# Query data dari koleksi
cursor = collection.find()

# Konversi cursor ke list dan kemudian ke DataFrame
data = list(cursor)
df = pd.DataFrame(data)

# Menutup koneksi
client.close()

In [5]:
df

_id                 channelId      videoId  \
0      66515ed9d882d1f847df446f  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
1      66515ed9d882d1f847df4470  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
2      66515ed9d882d1f847df4471  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
3      66515ed9d882d1f847df4472  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
4      66515ed9d882d1f847df4473  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
...                         ...                       ...          ...   
70130  66515f99d882d1f847e05661  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
70131  66515f99d882d1f847e05662  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
70132  66515f99d882d1f847e05663  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
70133  66515f99d882d1f847e05664  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   
70134  66515f99d882d1f847e05665  UC-z1ubmS-iHMVr83EEwRS2g  1ZIGLm5cuDo   

                                             textDisplay  \
0      Rewind paling bnyk memori soalnya di jaman itu...   
1      <a href="https://www.youtube.com/watch?v=1ZIGL...   
2            Seru banget liat ulang&quot; dari yang 2023   
3                                  Haii aku dari 2024🥹🇮🇩   
4                                         Halo dari 2024   
...                                                  ...   
70130                                           pertamax   
70131                                               Firs   
70132                                            pertama   
70133                                            KEREEEN   
70134                                              first   

                                            textOriginal  \
0      Rewind paling bnyk memori soalnya di jaman itu...   
1                                     3:30 BeaconCream🗿🗿   
2                 Seru banget liat ulang" dari yang 2023   
3                                  Haii aku dari 2024🥹🇮🇩   
4                                         Halo dari 2024   
...                                                  ...   
70130                                           pertamax   
70131                                               Firs   
70132                                            pertama   
70133                                            KEREEEN   
70134                                              first   

                     authorDisplayName  \
0                          @Fiammawan1   
1                       @PemainRoblox2   
2                             @Zan-too   
3             @adindatiaraagustine6028   
4                    @rnggaakhaira6602   
...                                ...   
70130  @lalumuhammadnaveenbomantar1066   
70131         @keisyanoviansyahnur2857   
70132                  @donylukito1058   
70133                         @fjr1746   
70134              @yustrisiadinda7687   

                                   authorProfileImageUrl  \
0      https://yt3.ggpht.com/_e9Fqq5cqPF2-9fSB3Uuu3-H...   
1      https://yt3.ggpht.com/ruCvUZnnBNROJMZFxR7eUxND...   
2      https://yt3.ggpht.com/ytc/AIdro_lqybv1ymmQZzV-...   
3      https://yt3.ggpht.com/ytc/AIdro_nZTGo3ibibGPIc...   
4      https://yt3.ggpht.com/ytc/AIdro_miy4I4tWQShzVX...   
...                                                  ...   
70130  https://yt3.ggpht.com/ytc/AIdro_lThIcUkdURy36C...   
70131  https://yt3.ggpht.com/ytc/AIdro_mNGnEcqEDgIssF...   
70132  https://yt3.ggpht.com/ytc/AIdro_kZXf00HgnNSZBH...   
70133  https://yt3.ggpht.com/ytc/AIdro_mlJRhIYHmv0vHd...   
70134  https://yt3.ggpht.com/ytc/AIdro_n04FDP-09gmUlz...   

                                        authorChannelUrl  \
0                     http://www.youtube.com/@Fiammawan1   
1                  http://www.youtube.com/@PemainRoblox2   
2                        http://www.youtube.com/@Zan-too   
3        http://www.youtube.com/@adindatiaraagustine6028   
4               http://www.youtube.com/@rnggaakhaira6602   
...                                                  ...   
70130  http://www.youtube.com/@lalumuhammadnaveenbo

#### Menghapus data text yang duplikat

In [6]:
df = df.drop_duplicates(subset=['textOriginal'])

df.shape

(65632, 14)

#### Menghapus data author yang duplikat

In [7]:
df = df.drop_duplicates(subset=['authorChannelUrl'])

df.shape

(57706, 14)

#### Menghilangkan atribut yang tidak diperlukan

Karena hanya menganalisis komentar, maka atribut yang disisakan hanya _id dan textOriginal

In [8]:
df = df[['_id', 'textOriginal']]
df

_id  \
0      66515ed9d882d1f847df446f   
1      66515ed9d882d1f847df4470   
2      66515ed9d882d1f847df4471   
3      66515ed9d882d1f847df4472   
4      66515ed9d882d1f847df4473   
...                         ...   
70118  66515f99d882d1f847e05655   
70122  66515f99d882d1f847e05659   
70128  66515f99d882d1f847e0565f   
70131  66515f99d882d1f847e05662   
70133  66515f99d882d1f847e05664   

                                            textOriginal  
0      Rewind paling bnyk memori soalnya di jaman itu...  
1                                     3:30 BeaconCream🗿🗿  
2                 Seru banget liat ulang" dari yang 2023  
3                                  Haii aku dari 2024🥹🇮🇩  
4                                         Halo dari 2024  
...                                                  ...  
70118                                                  t  
70122                                               firt  
70128                                           firstttt  
70131                                               Firs  
70133                                            KEREEEN  

[57706 rows x 2 columns]

#### Menghapus nilai yang kosong

In [9]:
df = df.dropna()

# Melihat data yang kosong
df.isnull().sum()

_id             0
textOriginal    0
dtype: int64

Dari yang awalnya berjumlah 70135 row berkurang menjadi 57706 row

#### Mengimpan Clean Comments ke MongoDB Local

In [10]:
# Koneksi ke MongoDB
client = MongoClient(local_url)
db = client[database_name]
collection = db[collection_name]

# Konversi DataFrame ke bentuk dictionary
records = df.to_dict(orient='records')

# Menyisipkan data ke MongoDB
collection.insert_many(records)

print("Data berhasil diimpor ke MongoDB Local.")

Data berhasil diimpor ke MongoDB Local.


#### Migrasi dari MongoDB Local ke MongoDB Atlas

In [11]:
# Koneksi ke MongoDB lokal
local_client = MongoClient(local_url)
local_db = local_client[database_name]
local_collection = local_db[collection_name]

# Koneksi ke MongoDB Atlas
atlas_client = MongoClient(mongodb_url)
atlas_db = atlas_client[database_name]
atlas_collection = atlas_db[collection_name]

# Ambil semua dokumen dari koleksi MongoDB lokal
documents = list(local_collection.find())

# Masukkan dokumen ke MongoDB Atlas
if documents:
    atlas_collection.insert_many(documents)
    print(f'{len(documents)} dokumen berhasil dipindahkan dari MongoDB lokal ke MongoDB Atlas.')
else:
    print('Tidak ada dokumen yang ditemukan di MongoDB lokal.')

# Tutup koneksi
local_client.close()
atlas_client.close()

KeyboardInterrupt: 